checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -534,6 +540,28 @@ public Void visitNewArray(NewArrayTree node, AnnotatedTypeMirror type) {
       }
       return null;
     }
+
+    @Override
+    public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror type) {
+      if (TreeUtils.isMethodInvocation(tree, copyOfMethods, processingEnv)) {
+        List<? extends ExpressionTree> args = tree.getArguments();
+        ExpressionTree lengthArg = args.get(1);
+        if (TreeUtils.isArrayLengthAccess(lengthArg)) {
+          // TODO: This syntactic test may not be not correct if the array expression has a side
```

Comment:
It seems unlikely that the side effect would impact the length of the array, so I'm fine leaving this TODO.